addons/viewrestriction/XEH_postInitClient.sqf

```diff
@@ -0,0 +1,104 @@
+#include "script_component.hpp"
+
+// Exit on Server and Headless
+if !(hasInterface) exitWith {};
+
+// Exit if third person view is not available
+if !(difficultyEnabled "3rdPersonView") exitWith {};
+
+["SettingsInitialized", {
+    // Exit if module is disabled
+    if (GVAR(mode) == 0) exitWith {};
+
+    // Start proper PFH based on mode (separated from one PFH with many conditions to improve performance)
+
+    // Conditions with cameraView results in visible transitions, ongoing use of switchCamera does not
+    // UAVs break without cameraView condition (control returns to player object, camera is left on UAV object)
+    // UAVs are therefore left out of this module until fixed or suitable work-around is found
+    #define IS_UAV (cameraOn isKindOf "UAV" || {cameraOn isKindOf "UAV_01_base_F"})
+
+    // FirstPerson
+    if (GVAR(mode) == 1) exitWith {
+        [{
+            if (isNull ACE_player || {!alive ACE_player}) exitWith {};
+
+            if (IS_UAV) exitWith {
+                // Reference comment in XEH_postInitClient.sqf, beginning of "SettingsInitialized" Event Handler
+            };
+            cameraOn switchCamera "Internal";
+        }, 0, []] call cba_fnc_addPerFrameHandler;
+    };
+
+    // ThirdPerson
+    if (GVAR(mode) == 2) exitWith {
+        [{
+            if (isNull ACE_player || {!alive ACE_player}) exitWith {};
+
+            if (IS_UAV) exitWith {
+                // Reference comment in XEH_postInitClient.sqf, beginning of "SettingsInitialized" Event Handler
+            };
+            cameraOn switchCamera "External";
+        }, 0, []] call cba_fnc_addPerFrameHandler;
+    };
+
+    // Selective
+    if (GVAR(mode) == 3) exitWith {
+        // Exit if all Selective Modes are Disabled
+        if (GVAR(modeSelectiveFoot) == 0 && {GVAR(modeSelectiveVehicle) == 0} && {GVAR(modeSelectiveAir) == 0} && {GVAR(modeSelectiveUAV) == 0}) exitWith {diag_log "selective disabled due to submodes"};
+
+        [{
+            if (isNull ACE_player || {!alive ACE_player}) exitWith {};
+
+            // Foot
+            if (cameraOn isKindOf "CAManBase") exitWith {
+                if (GVAR(modeSelectiveFoot) == 1) exitWith {
+                    cameraOn switchCamera "Internal";
+                };
+                if (GVAR(modeSelectiveFoot) == 2) exitWith {
+                    cameraOn switchCamera "External";
+                };
+            };
+
+            // Land Vehicles
+            if (cameraOn isKindOf "LandVehicle") exitWith {
+                if (GVAR(modeSelectiveLand) == 1) exitWith {
+                    cameraOn switchCamera "Internal";
```

Comment:
I have tested it all in Editor, but didn't get a chance to do it in multiplayer yet, though it shouldn't have any impact at all, all of it is run client-side only.

Settings are applied from via Settings Framework (ACE_Settings or module), they are not client-side settings, so it shouldn't matter if it does for all or not. Additionally BI wiki says:
> If you switch to a unit in a vehicle, this command uses the correct turret.

`cameraOn` returns the object so either vehicle player or player directly, which would be an alternative.